### ISSUE TITLE
Demo lighty.io project for SpringBoot fails on start

### ISSUE THREAD

**Issue by rhythmicdevil** [PantheonTechnologies/lighty-core#253]
Hi,
I cloned the repo and imported the SpringBoot lighty.io demo into Eclipse.
For some reason the Spring Boot Starter Parent cant be found. I get the following error.

```
For artifact {io.lighty.core:lighty-controller-spring-di:null:jar}: The version cannot be empty. (org.apache.maven.plugins:maven-resources-plugin:3.0.2:resources:default-resources:process-resources)
```

Maven Version:
```
$ mvn -v
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T15:00:29-04:00)
Maven home: /usr/local/src/apache-maven
Java version: 1.8.0_172, vendor: Oracle Corporation, runtime: /opt/eti/java/jdk1.8.0_172-x64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-957.21.2.el7.x86_64", arch: "amd64", family: "unix"

```

Can you please advise me on how to resolve this?

**Comment by jveverka** [comment]
Hi Steven
Thank you for your interest in lighty.io. After cloning lighty.io repository, you should build the project running `mvn clean install` this will generate yang java bindings used by some lighty.io projects.
After that open in eclipse (I tested eclipse for java EE developers) root lighty-core project and go inside lighty-controller-springboot-netconf project and run MainApp.java as Java application. This will start springboot demo without problems. Do not open only lighty-controller-springboot-netconf project, because it will be missing other dependencies.

**Comment by rhythmicdevil** [comment]
Juraj,
thank you for the speedy response. I did as stated and it built successfully. I then imported the lighty-controller-springboot-netconf project and started it up. It runs as expected. 

```
2019-06-13 12:29:10.027  INFO 28745 --- [pool-4-thread-1] i.l.c.c.impl.LightyControllerImpl        : Lighty controller started in 4248.0674ms
2019-06-13 12:29:10.027  INFO 28745 --- [pool-4-thread-1] i.l.c.c.api.AbstractLightyModule         : LightyModule LightyControllerImpl started.
2019-06-13 12:29:10.028  INFO 28745 --- [           main] i.l.c.c.s.config.LightyConfiguration     : LightyController Core started
2019-06-13 12:29:10.458  INFO 28745 --- [           main] i.l.c.c.api.AbstractLightyModule         : Submitted start of LightyModule NetconfTopologyPlugin.
2019-06-13 12:29:10.692  INFO 28745 --- [pool-9-thread-1] i.l.c.c.api.AbstractLightyModule         : LightyModule NetconfTopologyPlugin started.
```

If I may continue this conversation. The company I work for is evaluating the lighty.io for integration into our systems. We have existing spring boot stuff, and infrastructure (Artifactory) already setup. Having to replace or modify our settings.xml file is a little unwieldy. This might only be required for running the demos, but I am not sure. 

Basically we want to develop a spring boot app that uses lighty.io. We plan on developing northbound and southbound plugins for it, as well as yang models and netconf that we might need. 

We are unsure on how to start. Is there some documentation that might provide information on how to go about this? I looked the documentation [here ](https://lighty.io/lighty-io-sdn-made-easy/)but it seems out of date. 

Thanks for any help and advice you can offer. We look forward to seeing what lighty.io can do.

**Comment by jveverka** [comment]
Hi Steven
Here are some answers below, I suggest that we continue this discussion in separate email thread.
* modify settings.xml is one way, another possibility is to include profile in your pom.xml file 
* if you want to develop your own north / southbound plugins, you should start with yang yang models
* [this is how](https://github.com/PantheonTechnologies/lighty-core/blob/10.0.x/lighty-models/README.md) you create and use your own yang model in project 
* than you implement java code for north / south bound plugin
* resulting architecture depends on use case you need to implement into lighty.io controller.

**Comment by rhythmicdevil** [comment]
Thank you very much for the response. I will be happy to continue this discussion in email with you and our development group.
